x-pack/plugins/uptime/server/lib/requests/get_monitor_availability.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { UMElasticsearchQueryFn } from '../adapters';
+import { GetMonitorAvailabilityParams } from '../../../common/runtime_types';
+
+export interface AvailabilityKey {
+  monitorId: string;
+  location: string;
+  name: string;
+  url: string;
+}
+
+export interface GetMonitorAvailabilityResult {
+  monitorId: string;
+  location: string;
+  name: string;
+  url: string;
+  up: number;
+  down: number;
+  availabilityRatio: number | null;
+}
+
+export const formatBuckets = async (buckets: any[]): Promise<GetMonitorAvailabilityResult[]> => {
+  return buckets.map(({ key, up_sum, down_sum, ratio }: any) => ({
+    ...key,
+    up: up_sum.value,
+    down: down_sum.value,
+    availabilityRatio: ratio.value,
+  }));
+};
+
+export const getMonitorAvailability: UMElasticsearchQueryFn<
+  GetMonitorAvailabilityParams,
+  GetMonitorAvailabilityResult[]
+> = async ({ callES, dynamicSettings, range, rangeUnit, threshold: thresholdString, filters }) => {
```

Comment:
As I understand it, we compile the filters into the alert, then pass them into here. I realize that we serialize the individual parameters as well, so I wonder why those aren't passed in instead? It seems advantageous to compile ES syntax as late as possible to give us maximum flexibility, and prevent us from being dependent on serialized data. If ES changes the way a query works, we won't be able to easily handle that if the query is serialized into the alert.

Reply:
I think it's effectively the same in either place; the filter values we serialize are stored in their pre-serialized state in the saved object. We could move the code here, but we'd be calling it from multiple ES request functions instead of once in the alert, and it would increase the number of dependencies the requests need in order to function.

Reply:
You are a 100% right. I thought it was generated on save, not on execution. ++ on keeping it the way it is